src/Lykke.Service.OperationsHistory.Core/Entities/HistoryLogEntryRepository.cs

```diff
@@ -96,15 +98,32 @@ public HistoryLogEntryRepository(INoSQLTableStorage<HistoryLogEntryEntity> table
         public async Task<IList<HistoryLogEntryEntity>> GetByDatesAsync(DateTime dateFrom, DateTime dateTo)
         {
             var rangeQuery = AzureStorageUtils.QueryGenerator<HistoryLogEntryEntity>.PartitionKeyOnly.BetweenQuery(
-                    HistoryLogEntryEntity.ByDate.GeneratePartitionKey(dateFrom), 
-                    HistoryLogEntryEntity.ByDate.GeneratePartitionKey(dateTo.AddDays(-1)),
-                    ToIntervalOption.IncludeTo);
+                HistoryLogEntryEntity.ByDate.GeneratePartitionKey(dateFrom),
+                HistoryLogEntryEntity.ByDate.GeneratePartitionKey(dateTo.AddDays(-1)),
+                ToIntervalOption.IncludeTo);
 
             var data = new List<HistoryLogEntryEntity>();
 
             await _tableStorage.ExecuteAsync(rangeQuery, chunk => data.AddRange(chunk));
 
             return data;
         }
+
+        public async Task<IList<HistoryLogEntryEntity>> GetByWalletsAsync(IEnumerable<string> walletIds)
+        {
+            var result = new List<HistoryLogEntryEntity>();
+
+            await Task.WhenAll(
+                walletIds.Select(x => _tableStorage.GetDataByChunksAsync(
```

Comment:
Her can be opened a lot of connections to Azure. Probably it is better to divide walletIds by batches